boa/src/object/internal_methods.rs

```diff
@@ -282,6 +282,52 @@ impl Object {
         })
     }
 
+    /// Essential internal method OwnPropertyKeys
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///
+    /// [spec](https://tc39.es/ecma262/#table-essential-internal-methods)
+    pub fn own_property_keys(&self) -> Vec<PropertyKey> {
+        let size = self.indexed_properties.len()
+            + self.string_properties.len()
+            + self.symbol_properties.len();
+        let mut prop_keys = Vec::with_capacity(size);
+
+        self.index_properties()
+            .for_each(|(p, _)| prop_keys.push(PropertyKey::from(*p)));
+        self.string_properties()
+            .for_each(|(p, _)| prop_keys.push(PropertyKey::from(p.as_str())));
+        self.symbol_properties()
+            .for_each(|(p, _)| prop_keys.push(PropertyKey::from(p.clone())));
+
+        prop_keys
```

Comment:
```suggestion
        self.keys().collect()
```
we can use the `keys` iterator method